### PR TITLE
Use mjolnir.js for event handling

### LINKED ISSUE #126: Replace hacky event handling
Get rid of hacky pointer event handlers in `EditableLayer`. Instead, use the new events exposed in deck.gl 6.3.

Some pseudocode of it working elsewhere:

```js
class MyLayer extends CompositeLayer {
  constructor(props) {
    // NOTE: hacky, we shouldn't inject callbacks in this way.
    // PS1: onDrag, onClick, onHover need to be methods of MyLayer instead of arrow functions
    // Otherwise they will become undefined in the following callbacks
    // => I don't know why :(
    // PS2: these injections need to be happened before super(props);
    props.onDrag = (info, event) => info.layer.onDrag(info, event);
    props.onClick = info => info.layer.onClick(info);
    props.onHover = info => info.layer.onHover(info);
  }
}
```

### LINKED ISSUE #169: Draw doesn't work on Safari and iOS
#### Describe the bug
When clicking on the map using an iOS device onLayerClick seems to fire, but points are not drawn.
#### Actual Result
No points drawn
#### Expected Result
Points to show up on map.
#### Reproduce Steps
Run map, with Nebula.gl on iOS device. When you try and draw something no shapes or points are dropped.
#### To Do List 
  - [x] Add label and assign to milestone
  - [x] Coding
  - [ ] Test

### LINKED ISSUE #181: EditableGeoJsonLayer can't add points on mobile
#### Describe the bug

On my android phone (in chrome) when I click edit on  <https://nebula.gl/#/> I can see different coloured dots on the midpoints of lines so that I can subdivide that line

#### Actual Result

I can add points on a desktop using a mouse but when I switch to mobile I can no longer add points using touch

#### Reproduce Steps

Here's my code sandbox
https://codesandbox.io/s/9qo3l476po

And the subsequent URL I used on my Android phone:

https://9qo3l476po.codesandbox.io/

#### To Do List 
  - [x] Add label and assign to milestone
  - [x] Coding
  - [ ] Test

### LINKED ISSUE #262: UX
#### Feature request
Add instructions on the exact UI/UX needed to add/remove/edit GeoJson Features.  

#### Background 
I am not sure if the current release has bugs or if the intended UX is just unclear.   But trying to add/remove/edit polygons or line strings doesn't seem to work correctly.   Often times a Polygon can't be closed with double clicking.   Grabbing vertexes is also problematic.  I am not sure if the user is supposed to press keys while clicking/dragging etc.

#### To Do List 
  - [ ] Add label and assign to milestone
  - [ ] Coding
  - [ ] Doc update
  - [ ] Test

### LINKED ISSUE #276: Remove event handling hack
The `EditableLayer` used to bind/unbind event handlers on every update:

https://github.com/uber/nebula.gl/blob/4ffcd762820b234d0bc5b5db72d173004782c658/src/lib/layers/editable-layer.js#L91

Because `this` i.e. the layer instance may change over time.

A better solution to this problem is to use the `layer.getCurrentLayer()` method:

```js
canvas.addEventListener(
  'pointermove',
  event => {
    const layer = this.getCurrentLayer(); // returns the latest layer instance
    this._onPointerMove.call(layer, event);
  }
);
```

This should work with registering/unregistering events only once on initialization/finalization.

#### To Do List 
  - [x] Add label and assign to milestone
  - [ ] Coding
  - [ ] Test

### PR DESCRIPTION
**Pull request by supersonicclay**
Refactor to use mjolnir's EventManager exposed from deck through layer context.

Breaking change: `PointerMoveEvent` no longer has an `isDragging` flag. Instead, use the new `handleDragging` event.

Fixes #126
Fixes #169
Fixes #181
Fixes #276
Fixes #262

### CODE REVIEW

File: modules/layers/src/layers/editable-layer.js [hunk 1]
```diff
@@ -40,118 +42,100 @@ export default class EditableLayer extends CompositeLayer {
...
+    });
+    eventManager.on('panmove', event => this._forwardEventToCurrentLayer('_onPanMove', event), {
+      // give nebula a higher priority so that it can stop propagation to deck.gl's map panning handlers
+      priority: 100
+    });
+    eventManager.on('panend', event => this._forwardEventToCurrentLayer('_onPanEnd', event));
+    eventManager.on('dblclick', event => this._forwardEventToCurrentLayer('_onDoubleClick', event));
   }
 
-  _removePointerHandlers() {
```
[the remark this answers is not in the document]
Comment (supersonicclay) [reply modules/layers/src/layers/editable-layer.js]: will finalizeState call the same layer instance as initializeState?

[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/layers/editable-layer.js]: No. You need to save the registered handlers in the layer state.

[the remark this answers is not in the document]
Comment (supersonicclay) [reply modules/layers/src/layers/editable-layer.js]: Got it. Nice catch. Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ _All notable changes to this project will be documented in this file. This proje
 
 <!-- INSERT HERE -->
 
+<!-- NEXT VERSION
```
Comment (georgios-uber) [inline CHANGELOG.md]: Don't use title/comments. It confuses the auto-generation script

Comment (supersonicclay) [reply CHANGELOG.md]: Yeah, that's just temporary until we run the auto-generation script.

**Pull request merged**